### DISCUSSION TITLE
InsightFace's build - SCRFD-10GF Face Detector

### DISCUSSION THREAD

**Discussion by mnts-i** [exadel-inc/CompreFace#903] [Q&A] [answered]
Newbie question incoming. I am new to the whole ML thing. I am using Insightface's custom build which according to the readme file uses the **RetinaFace** for face detection. This model has some problems detecting faces (especially when they wear masks covering their mouths).

Out of curiosity I created a test notebook on google colab where I used **InsightFace's FaceAnalysis** with **buffalo_l** model (https://github.com/deepinsight/insightface/tree/master/python-package) which uses **SCRFD-10GF** for detection and has no problem detecting the above mentioned faces.

My question is: Is there a (easy) way to build a new compreface-core image with the **SCRFD-10GF** model for detection? Will it mess up the recognition's embedding output (maybe it was trained with retinaface doing the detection)?

**Comment by IvanKurnosov** [comment, marked as answer]
Hi, 
Adding a new model for face detection is simple. You should put your model to Google Drive, then copy the last part of the link (should look like this: 1mh0QHmjKor_H0WLJdFDeGfzG-zNIHAyd). 
In CompreFace code, go to the embedding-calculator/src/services/facescan/plugins/insightface/insightface.py, find FaceDetector class and add your name (can be any) and the link to the model that you just copied to the beginning of the ml_models tuple.
Now this model should be used by default and you can build CompreFace the same way as you built it before.

Face recognition output will not change if detector return bboxes of the same size and location. You can check it by running two models on the same face. I expect that it should be fine because both models are from Insightface so they should be trained with the same alignment settings.
